### ISSUE TITLE
Universal build fails on non-ARM64 hardware, and M1 target DMG is 'damaged'

### ISSUE THREAD

**Issue by Webreaper** [electron-userland/electron-builder#5850]
I've got an app which I build hosted in an Electron container, to give me some native/filesystem interop. It works well.

However, for some reason, if I build the M1/ARM64 version (or a Universal build) on GitHub actions, I get a 'damaged' DMG file (for the ARM64 build) and an odd Javascript/Node error for the universal build:

```
TypeError [ERR_INVALID_CALLBACK]: Callback must be a function
at makeCallback (fs.js:136:11)
at rmdir (fs.js:671:14)
at Promise (internal/util.js:274:30)
at new Promise (<anonymous>)
at rmdir (internal/util.js:273:12)
at MacPackager.doPack (/Users/runner/work/Damselfly/Damselfly/Damselfly.Desktop/node_modules/app-builder-lib/src/macPackager.ts:123:15)

```
You can see the build failure for the Universal version [here](https://github.com/Webreaper/Damselfly/runs/2486907712?check_suite_focus=true#step:4:34). The ARM64 build completes, but the resulting App fails to start, with MacOS saying it's 'damaged'.

Building locally on my M1 MacBook pro works fine. My hunch is that the GitHub Actions runners aren't MacOS ARM64 hardware, so can't build the M1 versions. But that's weird, because my understanding is that electron-builder should be able to build ARM64/Universal targets on an Intel Mac (as per [this issue](https://github.com/electron-userland/electron-builder/issues/5689). I'm sure it used to work about 3-4 weeks ago, and I haven't seen any references to Electron build needing to run on M1 hardware to build M1 targets.

I'm using Electron 11.0.0, electron-builder 22.10.4, node 10.24.1. The OS that I'm running this on is 11.4 beta, on an M1 Macbook pro.

**Comment by mmaietta** [comment]
Intel-based Macs can still build arm64 versions of your app, and universal builds as well when adding `lipo` for creating fat universal binaries. electron-builder uses @electron/universal package for accomplishing that.

Re: GitHub action failure. Pretty sure `promisify` is where it's failing.
https://github.com/electron-userland/electron-builder/blob/6237e670a8ddd99ec8c076db86361c06c04eedd4/packages/app-builder-lib/src/macPackager.ts#L131-L133

Would you be willing to try node 12? And while you're at it, update to `22.10.5`?

**Comment by Webreaper** [comment]
Thanks. Those upgrades solved the build issue for the Universal binary, so that's one thing solved. However, now I have two binaries - Universal and ARM64 - that both fail to launch, because MacOS thinks they're 'damaged':

`“Damselfly” is damaged and can’t be opened. You should move it to the Bin.`

[Logs here](https://github.com/Webreaper/Damselfly/runs/2489684194?check_suite_focus=true#step:4:21).

**Comment by mmaietta** [comment]
>   • skipped macOS application code signing  reason=cannot find valid "Developer ID Application" identity or custom non-Apple code signing certificate, see https://electron.build/code-signing allIdentities=

I think Big Sur and/or M1 require signing to be present, otherwise, it'll say it's damaged. Do you have the proper signing identity being provided?

Not sure if the "damaged" issue is occurring for all users or just for your local computer. If the latter, then this could help https://apple.stackexchange.com/a/372208

**Comment by nurpax** [comment]
The same build steps work for x86 though. At least in my tests, I don’t sign for either x86 or ARM. 

In my case I built on x86 mac and shared the dmg to another person who tried it on M1.

**Comment by Webreaper** [comment]
> I think Big Sur and/or M1 require signing to be present, otherwise, it'll say it's damaged. Do you have the proper signing identity being provided?


This is not correct. If you don't sign, the first time you open an app OSX will prompt that it's from an unverified publisher, but you can continue to run it anyway. I've never signed this app, and the x64 version works fine on Intel and M1 Macs. The Universal one is now working too. The M1 version works fine if I build it on my M1 mac, it just doesn't work (and gives the damaged error) if I build it on an Intel mac in Github Action's pipelines. 

> Not sure if the "damaged" issue is occurring for all users or just for your local computer.

It occurs for all users.

**Comment by mmaietta** [comment]
> Do you have any anecdotal examples of people who can successfully build arm64 electron apps on Intel Macs?

Build for arm64 on an Intel-based mac? I already can on electron-builder 22.10.5 + node 12, been doing so for quite some time now.

**Comment by Webreaper** [comment]
Yes, I used to be able to do this as well, but something has broken recently (which is, if I may be so bold, why I raised this issue...).

**Comment by mmaietta** [comment]
Your OP shows version 22.10.4. You say it broke recently, are you reporting an issue for 22.11.1?

**Comment by Webreaper** [comment]
In my second post I mentioned that I tried the upgrade you suggested in your first. Didn't realise there's an even newer version. I'll try it - are there specific fixes in that ver that are likely to help?

**Comment by Webreaper** [comment]
Okay, tried with 22.11.1 here: https://github.com/Webreaper/Damselfly/actions/runs/825677557 - if you download the Desktop-Dist.zip, you'll see the Universal DMG works, but the M1 (ARM64) DMG is damaged. Are you able to look at the ARM64 package and see why it's damaged, in case that'll give a clue what's broken in the build?

**Comment by mmaietta** [comment]
Can you try 22.11.4? We recently updated to use fs/promises

**Comment by Webreaper** [comment]
The universal build works with 22.11.4. I can no longer repro the problem with ARM64 builds, because I've removed it from my project (I only build universal now).

**Comment by Billiam** [comment]
I'm running into this issue as well with 22.11.7.

Building on github actions in a mac env (which should be using intel), I'm unable to create a non-damaged arm64 build. No issues with non-m1 mac builds.

**Comment by alectrocute** [comment]
+1. Stuck here currently. Anybody have further advice? Thanks!

**Comment by Webreaper** [comment]
The only advice I can give is to use github actions to build on a non-M1 host, and/or build universal. That's what I did to workaround it. Hoped the electron team might have fixed it by now though!

**Comment by mmaietta** [comment]
Would anyone be willing to try this patch on v22.11.10? I'm curious if the M1 devkits had a limitation that the production devices don't (or no longer have).
`dmg-builder+22.11.10.patch`
```
diff --git a/node_modules/dmg-builder/out/dmg.js b/node_modules/dmg-builder/out/dmg.js
index 0c479c8..ebf4520 100644
--- a/node_modules/dmg-builder/out/dmg.js
+++ b/node_modules/dmg-builder/out/dmg.js
@@ -173,13 +173,7 @@ async function createStageDmg(tempDmg, appPath, volumeName) {
     if (builder_util_1.log.isDebugEnabled) {
         imageArgs.push("-debug");
     }
-    let filesystem = ["HFS+", "-fsargs", "-c c=64,a=16,e=16"];
-    if (process.arch === "arm64") {
-        // Apple Silicon `hdiutil` dropped support for HFS+, so we force the latest type
-        // https://github.com/electron-userland/electron-builder/issues/4606
-        filesystem = ["APFS"];
-        builder_util_1.log.warn(null, "Detected arm64 process, HFS+ is unavailable. Creating dmg with APFS - supports Mac OSX 10.12+");
-    }
+    const filesystem = ["HFS+"];
     imageArgs.push("-fs", ...filesystem);
     imageArgs.push(tempDmg);
     // The reason for retrying up to ten times is that hdiutil create in some cases fail to unmount due to "resource busy".
```

**Comment by Julusian** [comment]
Try running `sudo xattr -r -d com.apple.quarantine something.app` for your built app.

From what I found when looking into arm builds a few weeks ago is that must be signed or they claim to be damaged. That command above should remove the flag indicating it is untrusted and will allow it to run.

The reason the builds work when produced on M1 is because they were built on that machine. If you were to download the dmg over http, they will also be reported as damaged.

I havent tried universal binaries, so I don't know how they behave.

**Comment by Julusian** [comment]
For reference, read the first paragraph under 'Code Signing' at https://developer.apple.com/documentation/macos-release-notes/macos-big-sur-11_0_1-universal-apps-release-notes/

**Comment by Webreaper** [comment]
Mac apps don't need to be signed (it's only a requirement if you're pushing to the AppStore). If you don't sign them, the first time you launch you'll see a dialog confirming you're happy to use an unsigned app, but they don't come up as 'damaged'. I've never signed any of my apps, and they've worked on OSX just fine. 

The universal builds that I'm creating also work fine too. So this isn't a code-signing issue; it's a problem with the DMG that electron-build is producing.

**Comment by Julusian** [comment]
@Webreaper if you read the link I provided, you will see this is no longer the case.
Once I setup signing, apps were no longer 'damaged' and ran fine

**Comment by alectrocute** [comment]
My issue ended up being `@electron/universal` improperly `lipo`'ing native node modules with `x64` and `x64` instead of `x64` and `arm64` - resulting in broken `arm64` builds. I'll make a ticket over there eventually. Spent so many hours on this...

**Comment by alectrocute** [comment]
@Webreaper I'm sitting in front of an M1 Mac right now and can confirm without a doubt that codesigning **is** required. An unsigned app will simply not run. On my Intel Mac, I'm able to still run unsigned apps albeit with a warning dialog. Refer to my comment above perhaps if you're using native modules.

**Comment by Webreaper** [comment]
I'm sitting in front of my M1 MacBook Pro and my universal builds are running just fine without being signed. My dotnet apps also run fine without being signed. You get prompted to allow them to run, but you can run them.

**Comment by Billiam** [comment]
> I'm sitting in front of my M1 MacBook Pro and my universal builds are running just fine without being signed

In the case of a universal build appearing to work, could it be that the m1 machines are running the non-arm version?

**Comment by Webreaper** [comment]
> In the case of a universal build appearing to work, could it be that the m1 machines are running the non-arm version?

How is that possible? If Macos identifies them as universal, it must execute the arm64 package.

**Comment by alectrocute** [comment]
> I'm sitting in front of my M1 MacBook Pro and my universal builds are running just fine without being signed. My dotnet apps also run fine without being signed. You get prompted to allow them to run, but you can run them.

If it’s the same machine you’ve built it on, then AFAIK it’ll let you run it. Try building on a separate machine and then trying running the app on your arm64 MBP.

**Comment by jamezrin** [comment]
We are still experiencing this issue over at https://github.com/notion-enhancer/notion-repackaged (tracked here https://github.com/notion-enhancer/notion-repackaged/issues/9), if there is anything we can do to help debug this issue, let me know.

**Comment by mmaietta** [comment]
Is this an error specifically in Github Actions and/or elsewhere? I'm unable to reproduce "DMG is damaged" error message
I've attempted the following scenarios with packing signed apps:

Built on my Intel mac, transferred to M1.
- x64 dmg ✅ 
- arm64 dmg ✅ 
- universal dmg ✅ 

Built on my M1, transferred to Intel.
- x64 dmg ✅ 
- arm64 dmg ✅ 
- universal dmg ✅ 

DMG opens in all cases.

This is my test repo https://github.com/mmaietta/electron-builder-test

**Comment by jamezrin** [comment]
> Is this an error specifically in Github Actions and/or elsewhere? I'm unable to reproduce "DMG is damaged" error message
> I've attempted the following scenarios with packing signed apps:
> 
> Built on my Intel mac, transferred to M1.
> 
> * x64 dmg white_check_mark
> * arm64 dmg white_check_mark
> * universal dmg white_check_mark
> 
> Built on my M1, transferred to Intel.
> 
> * x64 dmg white_check_mark
> * arm64 dmg white_check_mark
> * universal dmg white_check_mark
> 
> DMG opens in all cases.
> 
> This is my test repo https://github.com/mmaietta/electron-builder-test

Your test repo is not accessible, is it private?

Given the mix of experiences in this issue, this error might happen when not signing the app **or/and** using native binaries.

@alectrocute gave a lot of insight into that:

> My issue ended up being `@electron/universal` improperly `lipo`'ing native node modules with `x64` and `x64` instead of `x64` and `arm64` - resulting in broken `arm64` builds. I imagine if I were building on an M1 Mac, I'd have this problem in reverse. I'll make a ticket over on `@electron/universal` eventually. I also had to use the VSCode team's fork, but that's another story. Spent so many hours on this...
> 
> I had to manually create fat modules with `lipo` for every single module in my project, for example after using `electron-rebuild` to generate arch-specific binaries:
> 
> ```shell
> lipo ./node_modules/sqlite3/bin/darwin-arm64-85/sqlite3.node ./node_modules/sqlite3/bin/darwin-x64-85/sqlite3.node -create -output ./node_modules/sqlite3/build/Release/node_sqlite3.node;
> ```
> 
> Why is this relevant here? I got an "App is damaged" message and "You do not have permission to open this app" errors while trying to open my Universal app with `x64`-only `.node` modules. Could be a clue for someone here looking for an answer if all your apps are signed correctly (verified with `codesign -dvvv ./myApp.app`).

**Comment by Julusian** [comment]
@jamezrin could you running `sudo xattr -r -d com.apple.quarantine something.app` on the app then run it again?

**Comment by mmaietta** [comment]
Could someone please provide a minimum reproducible repo for this issue that I can look at?

Add this beforeBuild hook to force using `electron-rebuild` instead of `app-builder-bin` library.
`electron-builder.js` config. 
```
const electronRebuild = require('electron-rebuild');


    nodeGypRebuild: false,
    buildDependenciesFromSource: false,
    beforeBuild: async (context) => {
        const { appDir, electronVersion, arch } = context
        await electronRebuild.rebuild({ buildPath: appDir, electronVersion, arch })
        return false
    },
```

Alternatively, I think this would be `beforeBuild.js`
```
const electronRebuild = require('electron-rebuild');

module.exports = async (context) => {
        const { appDir, electronVersion, arch } = context
        await electronRebuild.rebuild({ buildPath: appDir, electronVersion, arch })
        return false
    }
```

**Comment by Tenpi** [comment]
Hello, I am experiencing this issue on an M1 Pro MBP running macOS Monterey 12.0.1. All of the apps built for the `arm64` architecture are reported as "damaged" when I download from github, but they open when built locally. 

Running `sudo xattr -r -d com.apple.quarantine "/Applications/myapp.app"` resolves the issue, but it's not user-friendly. I hope that this will get fixed soon, I'll unfortunately have to stick to building only for `x64` architecture.

**Comment by Hacksore** [comment]
Just wanted to add my experience with building `arm64` macOS builds via the `macos-latest` intel-based runners in CI.

I was getting the damaged error seen below but I added back codesigning to my CI builds and no 
longer get presented by at least that error anymore.

![image](https://user-images.githubusercontent.com/996134/145061429-c6cb420c-4420-4357-b915-f272184f682b.png)

While that's not a perfect solution for all do we have a way to run the gatekeeper attribute removal programmatically via a config that @Tenpi mentioned above?

Ultimately I think the best approach for anyone trying to distribute software for macOS you should sign and notarize it. Now there are instances like CI builds where notarizing doesn't make sense but you can at least sign them.

**Comment by Julusian** [comment]
@Hacksore 
> While that's not a perfect solution for all do we have a way to run the gatekeeper attribute removal programmatically via a config that @Tenpi mentioned above?

That won't work as its your local system adding that flag. So would only work if you distribute it with a shell script to be run first which will look really dodgy

> Ultimately I think the best approach for anyone trying to distribute software for macOS you should sign and notarize it. Now there are instances like CI builds where notarizing doesn't make sense but you can at least sign them.

Yes, this is exactly what I think the required solution is here. Because as I wrote above:
![image](https://user-images.githubusercontent.com/1327476/145792062-10f36c65-4a3b-4ef6-998a-8319296c4d07.png)
You might need to noratize too depending on the developer account, newer accounts are required to notarize: 
![image](https://user-images.githubusercontent.com/1327476/145792299-9ebf89b8-2855-40e7-a17f-85cf8ebd855b.png)

**Comment by Stanzilla** [comment]
Still an issue

**Comment by arevi** [comment]
Experiencing this same issue on my M1 Macbook building an application with electron-builder 23.1.0 on MacOS 12.4

**Comment by Kelier** [comment]
Still an issue

**Comment by jkroepke** [comment]
Mention that this issue is **not fixable** by electron. To avoid this error, you have to code-sign and notarize your application. 

To sign and notarize a native M1 application, you have to setup an Apple ID and enter the developer program of Apple (99$/year).

**Comment by Webreaper** [comment]
If that's the case, how come a universal apple Electron app works just fine?

**Comment by bojanczyk** [comment]
As mentioned by julusian, code signing is different for arm and intel macs. On an arm mac, if an app is not code signed, then the os will claim that the program is "damaged" and want to delete it. (On an intel mac, you will only get a warning that it is unsigned, which you can then overwrite.) If you do not have a pad Apple developer account, you can work around this by using something called "ad hoc" signing, which decorates the program with some kind of empty signature (they hyphen after -s stands for an ad hoc signature):

codesign --force --deep -s - name_of_your_app.app

This will eliminate the "damaged" message on arm macs, and revert to the intel situation, where the user can run the app once they have agreed to use it from an unidentified developer.

**Comment by divefox** [comment]
I fixed this problem by upgrade my mac osx version to 12+

**Comment by JerryI** [comment]
The problem is that most people are still using an older OSX. Is there any way to fix this without paying 99$/y to Apple?...

**Comment by henri9813** [comment]
Hello,

Does someone have a fix ? 

I'm absolutely not elated to electron, but I have the same issue ( for a Golang app ).

It's very disturbing but this thread is the most advanced one I found.

**Comment by JerryI** [comment]
> Hello,
> 
> Does someone have a fix ?
> 
> I'm absolutely not elated to electron, but I have the same issue ( for a Golang app ).
> 
> It's very disturbing but this thread is the most advanced one I found.

I believe its related more to Apple policy. You may still ship x86 Intel builds, which will run on Apple Silicon with limitations. 

I gave up and paid Apple for the license subscription. It solved all issues

**Comment by henri9813** [comment]
Okay,

Haha alright, thanks, I will ship intel based built. 

Best regards

**Comment by rahulbansal16** [comment]
@bojanczyk can it be done via windows as well?

**Comment by rahulbansal16** [comment]
> As mentioned by julusian, code signing is different for arm and intel macs. On an arm mac, if an app is not code signed, then the os will claim that the program is "damaged" and want to delete it. (On an intel mac, you will only get a warning that it is unsigned, which you can then overwrite.) If you do not have a pad Apple developer account, you can work around this by using something called "ad hoc" signing, which decorates the program with some kind of empty signature (the hyphen after -s stands for an ad hoc signature):
> 
> codesign --force --deep -s - name_of_your_app.app
> 
> This will eliminate the "damaged" message on arm macs, and revert to the intel situation, where the user can run the app once they have agreed to use it from an unidentified developer.
> 
> This issue is unrelated to GitHub actions.

@bojanczyk can it be done via windows as well.

**Comment by kthchew** [comment]
For the record #9007 should fix the "damaged" warning issue for macOS ARM64 builds without a signing identity in v26.0.13 and later (in prerelease atm). Of course you will never get past Gatekeeper without notarizing your app, but if you rebuild the app using that version or later you should get an "unverified" warning instead that can be bypassed in System Settings without the need for any Terminal commands, just like the situation on Intel.

**Comment by datgrog** [comment]
> For the record [#9007](https://github.com/electron-userland/electron-builder/pull/9007) should fix the "damaged" warning issue for macOS ARM64 builds without a signing identity in v26.0.13 and later (in prerelease atm). Of course you will never get past Gatekeeper without notarizing your app, but if you rebuild the app using that version or later you should get an "unverified" warning instead that can be bypassed in System Settings without the need for any Terminal commands, just like the situation on Intel.

@kthchew I'm testing with electron-builder@26.0.15 and using mac.identity: "-" for ad-hoc signing on an M2 MacBook Pro. The app builds and runs fine, but when I try to open it, macOS blocks it as expected. However, even after going to System Settings > Privacy & Security and clicking "Open Anyway", I still get the popup:

<img width="372" alt="Image" src="https://github.com/user-attachments/assets/ecf173bc-2ad5-49f1-bf17-22f6b0c08da7" />

“Check with the developer to make sure X_APP works with this version of macOS. You may need to reinstall the application.”

Is that still part of the expected behavior, or could there be something wrong with how the app is being packaged or signed? I thought after #9007, it should just show an “unverified developer” warning that could be bypassed. Appreciate any insights!

**Comment by JerryI** [comment]
> > For the record [#9007](https://github.com/electron-userland/electron-builder/pull/9007) should fix the "damaged" warning issue for macOS ARM64 builds without a signing identity in v26.0.13 and later (in prerelease atm). Of course you will never get past Gatekeeper without notarizing your app, but if you rebuild the app using that version or later you should get an "unverified" warning instead that can be bypassed in System Settings without the need for any Terminal commands, just like the situation on Intel.
> 
> @kthchew I'm testing with electron-builder@26.0.15 and using mac.identity: "-" for ad-hoc signing on an M2 MacBook Pro. The app builds and runs fine, but when I try to open it, macOS blocks it as expected. However, even after going to System Settings > Privacy & Security and clicking "Open Anyway", I still get the popup:
> 
> <img width="372" alt="Image" src="https://github.com/user-attachments/assets/ecf173bc-2ad5-49f1-bf17-22f6b0c08da7" />
> 
> “Check with the developer to make sure X_APP works with this version of macOS. You may need to reinstall the application.”
> 
> Is that still part of the expected behavior, or could there be something wrong with how the app is being packaged or signed? I thought after #9007, it should just show an “unverified developer” warning that could be bypassed. Appreciate any insights!

Did you sign it using Apple Developer account?

Error message vary slightly from OS versions (noticed on Sonoma). In general no one can bypass the signature check on Apple Silicon machines with a simple tweak from the settings menu.

**Comment by kthchew** [comment]
@datgrog That is not expected. The error seems to suggest that your app has passed the execution policy checks but is crashing on startup based on the ability to report this to Apple - if you open Console.app and look in the crash reports section, is there an entry that corresponds to your app?

**Comment by datgrog** [comment]
@kthchew many thanks for your help. It feels weird a bit because back when I was on `"electron-builder": "26.0.13"` arm64 would start normally or be flag damaged if served on my local/prod server http/https. When I updated to `"electron-builder": "26.0.15"` I notice the first time (only local) the “Open anyway” so I was like “Yay” but then facing the pop up show above. Since then every arm64 but trigger this popup, I’m not facing the “Open anyway” anymore. 
I tried

```bash
rm -rf node_modules dist out
bun run clean && electron-vite build && electron-builder --mac
```

My `electron-builder` config is targeting only `arm64` for mac:

```
mac:
  artifactName: '${productName}-${version}-${arch}.${ext}'
  target:
    - target: dmg
      arch:
        - arm64
  icon: icon.icns
  notarize: false // with these 3 lines or no => same result
  identity: '-'
  gatekeeperAssess: false
```

Despite this, I'm still encountering the same crash on launch https://pastebin.com/ZuzSrhYw

**Comment by kthchew** [comment]
@datgrog 

> Reason: tried: '/Applications/MedNoteSecure.app/Contents/Frameworks/Electron Framework.framework/Electron Framework' (code signature in <4C4C44B2-5555-3144-A131-DBFDE1E27027> '/Applications/MedNoteSecure.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework' not valid for use in process: mapping process and mapped file (non-platform) have different Team IDs)

I believe this is your issue. You're running up against a [hardened runtime](https://developer.apple.com/documentation/security/hardened-runtime) protection.

There's two ways you could fix this:

1. (better option) Add the [`com.apple.security.cs.disable-library-validation`](https://developer.apple.com/documentation/BundleResources/Entitlements/com.apple.security.cs.disable-library-validation) entitlement. Note that if you later sign your app with a real certificate, you should likely remove this entitlement. (ref: [electron-builder option](https://www.electron.build/mac#entitlements))
2. Disable hardened runtime entirely (ref: [electron-builder option](https://developer.apple.com/documentation/BundleResources/Entitlements/com.apple.security.cs.disable-library-validation))

The reason you didn't get this in 26.0.12 is that the hardened runtime on/off state is stored in the code signature. When there's no code signature, hardened runtime defaults to off and this check doesn't happen.

Out of curiosity, did you explicitly set either the hardened runtime or the entitlements options? I don't recall seeing this issue in my tests but looking at the docs the defaults don't seem suitable for ad-hoc signatures, so there might be room for improvement here. At the very least it might be worth warning if this configuration is detected.

**Comment by datgrog** [comment]
@kthchew thanks a ton for the explanation! I tried both fixes and yep it worked! Adding `<key>com.apple.security.cs.disable-library-validation</key>` on top of my existing entitlements did the trick so I will follow your advice. I set `hardenedRuntime` true explicitly in the conf. 

So I’m gonna stick with that for now. Here’s what my entitlements look like now:

```xml
<key>com.apple.security.cs.disable-library-validation</key>
<true/>
<key>com.apple.security.cs.allow-jit</key>
<true/>
<key>com.apple.security.cs.allow-unsigned-executable-memory</key>
<true/>
<key>com.apple.security.cs.allow-dyld-environment-variables</key>
<true/>
``` 
and my config
```yaml
mac:
  artifactName: '${productName}-${version}-${arch}.${ext}'
  target:
    - target: dmg
      arch:
        # - universal
        # - x64
        - arm64
  icon: icon.icns
  notarize: false # Whether to disable electron-builder’s @electron/notarize integration.
  gatekeeperAssess: false # Whether to let @electron/osx-sign validate the signing or not.
  identity: '-'
  hardenedRuntime: true
  entitlements: resources/build/entitlements.mac.plist
  entitlementsInherit: resources/build/entitlements.mac.plist
```

Really appreciate your help!! 💪
